specs/architecture/README.md

```diff
@@ -0,0 +1,95 @@
+* **Use only what you need - modular and extensible:** VDK aims to enable users to choose to use only the specific functionalities or modules they need and nothing more. And developers can extend the framework with new functionalities as the data ecosystem evolves.
+
+# Architecture
+
+## [System Context Diagrams](https://c4model.com/#SystemContextDiagram)
+
+Versatile Data Kit is conceptually composed of two independently usable components:
+- Control Service
+- Python-based SDK.
+
+
+The Control Service provides all the horizontal components required to configure and run [data jobs](https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job) on any cloud: the UI, and REST APIs for deployment, execution, properties and secrets. Also job scheduling, logging, alerting, monitoring etc. It is tightly coupled with Kubernetes and relies on Kubernetes ecosystem for many of its functionalities.
+
+> Note: all diagrams in the document are based on https://c4model.com
+
+![VDK Control Service System Context Diagram](https://github.com/vmware/versatile-data-kit/assets/2536458/84da99c8-2d2d-44d9-90a8-f0ea6ae5150f)
+
+
+<br><br>
+
+The VDK SDK consists of all user tools that assist in developing data jobs focusing on data ingestion and transformation.
+It can be installed locally on the user's machine to enable a local development experience.
+It follows plugin-oriented architecture where all functionality is built through plugins which enables reconstructing VDK SDK in any flavor depending on the organization's and users' needs.
+
+![VDK SDK System Context Diagram](https://github.com/vmware/versatile-data-kit/assets/2536458/8fab24c9-5d22-4f83-a41f-1a025f2edb4a)
+
+## VDK Control Service Container Diagram
+
+![VDK Control Service Container Diagram](https://github.com/vmware/versatile-data-kit/assets/2536458/94538ffb-e9a8-4a61-86ff-0a5b5d8a374a)
+
+Diving deeper into the VDK Control Service, the entry point from data practitioner's perspective is
+* VDK Frontend UI which is used for operating and monitoring of data jobs.
+* [VDK Rest API](https://iaclqhm5xk.execute-api.us-west-1.amazonaws.com/data-jobs/swagger-ui/index.html#/) (through VDK Control CLI or Notebook UI) which is used to deploy and configure data jobs.
+
+From operator's perspective:
+* Operators use provided [helm chart](https://github.com/vmware/versatile-data-kit/wiki/Installation#install-versatile-data-kit-control-service) to install, and configure the Control Service deployment and the needed data infrastructure.
+
+#### Rest API
+
+[Control Service Rest API](https://iaclqhm5xk.execute-api.us-west-1.amazonaws.com/data-jobs/swagger-ui/index.html) reflects the usual Development lifecycle:
+- Create a new data job when starting on a new use case.
+- Develop, run, and test the data job locally.
```

Comment:
I don't understand why this is part of the Rest API section?

Reply:
To be honest we copied it over from https://iaclqhm5xk.execute-api.us-west-1.amazonaws.com/data-jobs/swagger-ui/index.html  description

It is vital that they are similar.  But I can re-do it in both places. This description has not been updated since the initial version. 

If you forget what I've written, how would you describe the Control Service REST API , @dakodakov ?

Reply:
In my opinion the Control Service provides the means to deploy and manage data jobs... The creation/development and LOCAL runs are done through the vdk sdk.